lib/infrastructure/models/fx_model.dart

```diff
@@ -16,15 +17,15 @@ class FXModel {
       : id = 0,
         ts = DateTime.now(),
         value = 1,
-  decimals = 6,
-  name = 'ALGO',
-  unitname = 'ALGO',
-  iconUrl = null;
+        decimals = 6,
+        name = 'ALGO',
+        unitname = 'ALGO',
+        iconUrl = null;
 
   FXModel.fromJson(Map<String, dynamic> json, int docId)
       : id = docId,
-        ts = json['ts'].toDate(),
-        value = double.parse(json['value'].toString()),
+        ts = (json['ts'] as Object).toDate() ?? DateTime.now(),
```

Comment:
when can ts be null?

Reply:
This is for null safety, but we don't need, It fixed in the latest commit.

Reply:
null safety is needed if the value can be null. but in our app, ts can never be null. if ts becomes null, it means something went wrong somewhere else.